app/src/main/java/com/sdp/movemeet/backend/FirebaseInteraction.java

```diff
@@ -41,58 +41,58 @@
-     * Firestore and directly update corresponding TextViews.
-     * @param fStore Firebase Firestore reference
-     * @param userId The ID of the user
-     * @param textViewArray A TextView array containing the TextViews referencing user data (full
-     *                      name, email, phone number and profile description) to be updated
-     * @param activity The activity from which this function is called
-     * @return An array of TextViews containing the updated user data
-     */
-    public static TextView[] retrieveDataFromFirebase(FirebaseFirestore fStore, String userId, TextView[] textViewArray, Activity activity) {
-        DocumentReference documentReference = fStore.collection("users").document(userId);
-        documentReference.addSnapshotListener(activity, new EventListener<DocumentSnapshot>() {
-            @Override
-            public void onEvent(@Nullable DocumentSnapshot documentSnapshot, @Nullable FirebaseFirestoreException e) {
-                if (documentSnapshot == null) {
-                    //nothing
-                } else if (textViewArray.length == 4) {
-                    textViewArray[0].setText(documentSnapshot.getString("fullName"));
-                    textViewArray[1].setText(documentSnapshot.getString("email"));
-                    textViewArray[2].setText(documentSnapshot.getString("phone"));
-                    textViewArray[3].setText(documentSnapshot.getString("description"));
-                } else {
-                    textViewArray[0].setText(documentSnapshot.getString("fullName"));
-                    textViewArray[1].setText(documentSnapshot.getString("email"));
-                    textViewArray[2].setText(documentSnapshot.getString("phone"));
-                }
-            }
-        });
-
-        return textViewArray;
-    }
-
-    /**
-     * Create a HashMap containing the user data to be updated in Firebase Firestore. This function
-     * is used in {@link EditProfileActivity} to adjust the user data.
-     *
-     * @param profileFullName TextView corresponding to the full name of the user
-     * @param profileEmail TextView corresponding to the email of the user
-     * @param profilePhone TextView corresponding to the phone number of the user
-     * @param profileDescription TextView corresponding to the profile description set by the user
-     * @return A HashMap containing the updated user information
-     */
-    public static Map<String, Object> updateDataInFirebase(TextView profileFullName, TextView profileEmail, TextView profilePhone, TextView profileDescription) {
-        Map<String, Object> edited = new HashMap<>();
-        edited.put("fullName", profileFullName.getText().toString());
-        edited.put("email", profileEmail.getText().toString());
-        edited.put("phone", profilePhone.getText().toString());
-        edited.put("description", profileDescription.getText().toString());
+//    /**
```

Comment:
Why do you comment the firebase interaction ?

Reply:
This was in case something went wrong when I implemented the Firebase Storage abstraction (so that I could quickly go back to what we had previously in FirebaseInteraction, to refer to it). But, now, I was able to completely remove FirebaseInteraction! So the question is not even worth asking! 😂